ext/opcache/zend_file_cache.c

```diff
@@ -117,7 +117,7 @@ static int zend_file_cache_flock(int fd, int type)
 #define IS_SERIALIZED(ptr) \
 	((char*)(ptr) <= (char*)script->size)
 #define IS_UNSERIALIZED(ptr) \
-	(((char*)(ptr) >= (char*)script->mem && (char*)(ptr) < (char*)script->mem + script->size) || \
+	(((char*)(ptr) >= (char*)script->mem && (char*)(ptr) <= (char*)script->mem + script->size) || \
```

Comment:
This change looks wrong to me.

Reply:
I'm not sure why this problem showed up now, but the reason for this change is that if you have an arginfo with only a return type as the last allocation, and we store a pointer to `arginfo[1]`, then it will point to the end of the memory segment.

Reply:
Better, commit this separately with explanation comments.

Reply:
I've applied this separately in https://github.com/php/php-src/commit/23afc62080588f612fa6c5d0ea217564930dab3d. I believe the reason why it was not a problem before is that the main op_array was always located last, and it never has arg_info.